docs/source/additional_features.rst

```diff
@@ -24,7 +24,7 @@ They can be defined using the :py:func:`@dataclasses.dataclass
         plugins: List[str] = field(default_factory=list)
 
     test = Application("Testing...")  # OK
-    bad = Application("Testing...", "with plugin")  # Error: List[str] expected
+    bad = Application("Testing...", "with plugin")  # error: Argument 2 to "Application" has incompatible type "str"; expected "List[str]"
```

Comment:
I'd try to keep lines a bit shorter to avoid horizontal scrolling in most cases. Maybe keep them under 100 characters? I think that it's fine to shorten messages. For example leaving out `to "Application"` above seems reasonable. Also, it would be fine to split he comment into multiple lines. Maybe like this:

```
    bad = Application("Testing...", "with plugin")  # error: Argument 2 has incompatible type "str"; 
                                                    # expected "List[str]"
```